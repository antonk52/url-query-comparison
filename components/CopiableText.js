import {useState} from 'react';

const CopiableText  = ({value}) => {
    const [copied, setCopied] = useState(false);

    const handleClick = () => {
        if (value === null || value === undefined) {
            return;
        }
        navigator.clipboard.writeText(value).then(() => {
            setCopied(true);
            setTimeout(() => {
                setCopied(false);
            }, 1500);
        });
    };

    return (
        <span onClick={handleClick} className={copied && 'copied'}>
           {value || '-'}
            <style jsx>{`
                span {
                   display: inline-block;
                   max-width: 100%;
                   word-break: break-all;
                   word-wrap: break-word;
                   position: relative;
                   cursor: pointer;
                }
                
                .copied {
                    visibility: hidden;
                }
                
                .copied:after {
                    content: 'Copied!';
                    display: inline-block;
                    position: absolute;
                    left: 0;
                    top: 0;
                    color: #0076FF;
                    visibility: visible;
                
                    word-break: normal;
                    word-wrap: normal;
                    overflow: visible;
                }
           `}
           </style>
       </span>
    );
};

export default CopiableText;
