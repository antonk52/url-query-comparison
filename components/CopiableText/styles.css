button {
    display: inline-block;
    max-width: 100%;
    word-break: break-all;
    word-wrap: break-word;
    position: relative;
    cursor: pointer;

    border: none;
    text-decoration: none;
    user-select: none;
}

.copied {
    visibility: hidden;
}

.copied:after {
    content: 'Copied!';
    display: inline-block;
    position: absolute;
    left: 0;
    top: 0;
    color: #0366D6;
    visibility: visible;

    word-break: normal;
    word-wrap: normal;
    overflow: visible;
}
