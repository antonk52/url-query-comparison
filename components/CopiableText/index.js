import {useState} from 'react';
import styled from 'reshadow';

import styles from './styles.css';

const CopiableText  = ({value}) => {
    const [copied, setCopied] = useState(false);

    const handleClick = () => {
        if (value === null || value === undefined) {
            return;
        }

        if (navigator && navigator.clipboard) {
            navigator.clipboard.writeText(value).then(() => {
                setCopied(true);
                setTimeout(() => {
                    setCopied(false);
                }, 1500);
            });
        }
    };

    return styled(styles)(
        <button onClick={handleClick}>
            <text as="span" use:copied={copied}>{value || '-'}</text>
        </button>
    );
};

export default CopiableText;
