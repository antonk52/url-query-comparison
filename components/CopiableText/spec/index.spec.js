import React from 'react';
import {shallow} from 'enzyme';

import CopiableText from '../';

describe('CopiableText', () => {
    it('snapshot', () => {
        const wrapper = shallow(<CopiableText />);

        expect(wrapper).toMatchSnapshot();
    });
});
