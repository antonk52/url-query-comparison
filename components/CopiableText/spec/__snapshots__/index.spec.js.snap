// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopiableText snapshot 1`] = `
<button
  className="___button_40uj5_1"
  onClick={[Function]}
>
  <span
    className="___text_40uj5_1"
  >
    -
  </span>
</button>
`;
